File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let words = s.split(' ');
    let ans = '';

    for(let i=words.length-1; i>=0; i--){
        if(words[i] === '' ) continue;
        ans += words[i] + ' ';
    }

    return ans.trim();
    
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        stringstream str(s);
        vector<string> v;
        string ans, word;
        
        while(str>>word){
            v.push_back(word);
        }
        
        reverse(v.begin(), v.end());
        
        for(auto it: v){
            ans += " " + it;
        }
        
        ans.erase(0, 1);
        return ans;
    }
};